src/utils/getProducts.ts

```diff
@@ -0,0 +1,16 @@
+import axios from "axios";
+import { ProductType } from "@/types/product";
+
+const getProducts = async (id?: number): Promise<ProductType | ProductType[] | null> => {
+  const url = `https://dummyjson.com/products${id ? `/${id}` : ""}`;
+
+  try {
+    const res = await axios.get(url);
+    return id ? res.data : res.data.products;
+  } catch (error) {
+    console.error(error);
+    return null;
+  }
+};
+
```

Comment:
```suggestion
const getProducts = async (id?: number) => {
  const url = `https://dummyjson.com/products${id ? `/${id}` : ""}`;

  try {
    const res = await axios.get< ProductType[] | ProductType>(url);
    return id ? res.data : res.data.products;
  } catch (error) {
    console.error(error);
    return null;
  }
};

```

You can put ProductType into generics, no need to explicitly put return types into function signature, the'll be inferred, also I would split this function into 2 functions as it is clearly that these are 2 distinct endpoints `getAllProducts`  and `getProductById`, it kinda neglects SRP, overall it's acceptable